Source/cmake/OptionsCommon.cmake

```diff
@@ -154,7 +154,7 @@ if (USE_THIN_ARCHIVES)
 endif ()
 
 set(ENABLE_DEBUG_FISSION_DEFAULT OFF)
-if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+if (CMAKE_BUILD_TYPE STREQUAL "Debug" OR "${CMAKE_CXX_FLAGS_RELEASE}" MATCHES "-g")
```

Comment:
Hi, I think this won't work if build type is RelWithDebInfo or MinSizeRel or something custom, or if -g is passed via `CMAKE_CXX_FLAGS` rather than `CMAKE_CXX_FLAGS_RELEASE`. What CMake does is take `CMAKE_CXX_FLAGS` (which will default to using the environment variable `CXXFLAGS` if not specified), then append `CMAKE_CXX_FLAGS_<CONFIG>` ([reference)](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_FLAGS_CONFIG.html). So to reliably detect if a flag is passed, I think you need to check both `${CMAKE_CXX_FLAGS}` and `${CMAKE_CXX_FLAGS_$<CONFIG>}`. Who knows if that actually works, though.

(Also wonder if this would break distro find-debuginfo scripts, but the best way to find out is to proceed. ;)

Reply:
It works, I tested it, at least downstream.

Reply:
Actually downstream we don't build debug so we only care about release, that's why I used that variable. What do you think I should do to make it more general then? (I have trouble to properly understand your comment)

Reply:
Try `${CMAKE_BUILD_TYPE} STREQUAL "Debug" OR "${CMAKE_BUILD_TYPE} STREQUAL "RelWithDebInfo" OR ${CMAKE_CXX_FLAGS} MATCHES "-g" OR ${CMAKE_CXX_FLAGS_$<CONFIG>} MATCHES "-g"` (I'm hoping I have the syntax right for that generator expression...)